wagtail/core/tests/test_page_queryset.py

```diff
@@ -516,37 +516,48 @@ class TestSpecificQuery(TestCase):
 
     The fixture sets up a page structure like:
 
-    =========== =========================================
-    Type        Path
-    =========== =========================================
-    Page        /
-    Page        /home/
-    SimplePage  /home/about-us/
-    EventIndex  /home/events/
-    EventPage   /home/events/christmas/
-    EventPage   /home/events/someone-elses-event/
-    EventPage   /home/events/tentative-unpublished-event/
-    SimplePage  /home/other/
-    EventPage   /home/other/special-event/
-    =========== =========================================
+    ===================  =========================================
+    Type                 Path
+    ===================  =========================================
+    Page                 /
+    Page                 /home/
+    SimplePage           /home/about-us/
+    SimpleProxyPage      /home/simple-proxy-page/
+    SimpleProxyPageDeux  /home/simple-proxy-page-deux/
+    EventIndex           /home/events/
+    EventPage            /home/events/christmas/
+    EventPage            /home/events/someone-elses-event/
+    EventPage            /home/events/tentative-unpublished-event/
+    SimplePage           /home/other/
+    EventPage            /home/other/special-event/
+    ===================  =========================================
     """
-
     fixtures = ['test_specific.json']
 
-    def test_specific(self):
-        root = Page.objects.get(url_path='/home/')
+    def setUp(self):
+        self.root_page = Page.objects.get(url_path='/home/')
+        self.specific_live_descendants_of_root = [
+            Page.objects.get(url_path='/home/events/').specific,
+            Page.objects.get(url_path='/home/events/christmas/').specific,
+            Page.objects.get(url_path='/home/about-us/').specific,
+            Page.objects.get(url_path='/home/other/').specific,
+            Page.objects.get(url_path='/home/other/special-event/').specific,
+            Page.objects.get(url_path='/home/simple-proxy-page/').specific,
+            Page.objects.get(url_path='/home/simple-proxy-page-deux/').specific,
+        ]
 
+    def test_specific(self):
         with self.assertNumQueries(0):
             # The query should be lazy.
-            qs = root.get_descendants().specific()
+            qs = self.root_page.get_descendants().specific()
 
         with self.assertNumQueries(4):
-            # One query to get page type and ID, one query per page type:
-            # EventIndex, EventPage, SimplePage
+            # One query to get page type and ID, plus one query per concrete page type:
```

Comment:
I'm not very familiar with how this all works right now. Why are there two queries for SimplePage?

Reply:
Ah, this is just a typo in the comment (doh!) Those are the only 3 models that queries are made for (the other if to fetch page types and ids)

Reply:
Resolved in d5abca3